Sources/Sentry/SentryNSTimerFactory.m

```diff
@@ -1,11 +1,14 @@
 #import "SentryNSTimerFactory.h"
+#import "SentryInternalDefines.h"
 
 @implementation SentryNSTimerFactory
 
 - (NSTimer *)scheduledTimerWithTimeInterval:(NSTimeInterval)interval
                                     repeats:(BOOL)repeats
                                       block:(void (^)(NSTimer *timer))block
 {
+    SENTRY_ASSERT([NSThread isMainThread],
+        @"Timers must be scheduled from the main thread, or they may never fire.");
```

Comment:
Im curious where this is coming from? Never read anything about this.

Reply:
@armcknight, I think you once linked some Apple docs for that. Please add this as a comment.

Reply:
After looking for where I saw it, I learned a little bit more. It has to do with run loops and their modes. 

The original thing I saw that tipped me off to this is only found in an attribute on the declarations in NSTimer.h, not in any other docs: `+ (NSTimer *)scheduledTimerWithTimeInterval:(NSTimeInterval)ti target:(id)aTarget selector:(SEL)aSelector userInfo:(nullable id)userInfo repeats:(BOOL)yesOrNo NS_SWIFT_UNAVAILABLE_FROM_ASYNC("Timers scheduled in an async context may never fire.");`

I found a couple things talking about the run loop modes, which is most easily seen in this SO answer: https://stackoverflow.com/a/12173900. They link to this blog post: http://bynomial.com/blog/?p=67. I also found this one: https://danielemargutti.medium.com/the-secret-world-of-nstimer-708f508c9eb.

How should I word this? Something like: `According to the declaration in NSTimer.h, scheduled timers cannot be added from async contexts. They should only be scheduled from the main run loop.`?

Reply:
See my update, I simply point to the attribute on the declaration in NSTimer.h.

Reply:
I think you could also add the link as a code comment.